Code/Editor/ProcessInfo.cpp

```diff
@@ -108,14 +108,14 @@ void CProcessInfo::QueryMemInfo(ProcessMemInfo& meminfo)
     }
 #elif defined(AZ_PLATFORM_LINUX)
 
-    auto parseSize = [](const char* target, const AZStd::string_view& line, int* value)
+    auto parseSize = [](const char* target, const AZStd::string_view& line, int64& value)
     {
         if (line.starts_with(target))
         {
             auto substr = line.substr(strlen(target));
             if (!substr.empty())
             {
-                (*value) = (AZ::StringFunc::ToInt(substr.begin()) * 1024);
+                value = (aznumeric_cast<int64>(AZ::StringFunc::ToInt(substr.begin())) * 1024);
```

Comment:
question: Do we also need to cast (or use `ull`) for `1024` as it will just be a plain `int` as literals default to that? An extra cast might be worth it for completeness (not sure if it'll generate a warning or not, might be something to quickly check on compiler explorer with all the warnings cranked all the way up).

Reply:
hmm, looks kind of the same.

https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:10,endLineNumber:7,positionColumn:10,positionLineNumber:7,selectionStartColumn:10,selectionStartLineNumber:7,startColumn:10,startLineNumber:7),source:'//+Type+your+code+here,+or+load+an+example.%0A%23include+%3Ccstdint%3E%0Aint+square(uint64_t+a)+%7B%0A%0A++++int+b+%3D+a+*+1024%3B%0A++++int+e+%3D+a+*+1024ULL%3B%0A++++uint64_t+c+%3D+a+*+1024%3B%0A++++uint64_t+d+%3D+a+*+1024ull%3B%0A%0A++++return+a+*+c%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:compiler,i:(compiler:clang1400,filters:(b:'0',binary:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',selection:(endColumn:36,endLineNumber:8,positionColumn:36,positionLineNumber:8,selectionStartColumn:33,selectionStartLineNumber:8,startColumn:33,startLineNumber:8),source:1,tree:'1'),l:'5',n:'0',o:'x86-64+clang+14.0.0+(C%2B%2B,+Editor+%231,+Compiler+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4

Reply:
I believe C++ always converts the smaller type to the larger type in arithmetic operations. See here for example:
https://www.informit.com/articles/article.aspx?p=352857&seqNum=4 . (See the section "Conversions in Expressions")

Reply:
Thanks for confirming, looks good :+1: